code/optimize/src/androidTest/java/com/adobe/marketing/mobile/optimize/OptimizeFunctionalTests.java

```diff
@@ -939,8 +939,14 @@ public void testTrackPropositions_validPropositionInteractionsForDisplay() throw
         Map<String, Object> xdm = (Map<String, Object>) edgeRequestEventList.get(0).getEventData().get("xdm");
         Assert.assertEquals("decisioning.propositionDisplay", xdm.get("eventType"));
 
-        List<Map<String, Object>> propositionList = (List<Map<String, Object>>) ((Map<String, Object>) ((Map<String, Object>) xdm.get("_experience")).get("decisioning"))
-                .get("propositions");
+        Map<String, Object> experience =  (Map<String, Object>)xdm.get("_experience");
+        assertNotNull(experience);
+        Map<String, Object> decisioning =  (Map<String, Object>)experience.get("decisioning");
+        assertNotNull(decisioning);
+        Map<String, Object> propositionEventType = (Map<String, Object>)decisioning.get("propositionEventType");
+        assertNotNull(propositionEventType);
+        assertEquals(1, propositionEventType.get("display"));
+        List<Map<String, Object>> propositionList = (List<Map<String, Object>>)decisioning.get("propositions");
```

Comment:
nit: the lists and map objects defined below (https://github.com/adobe/aepsdk-optimize-android/pull/33/files#diff-d53c21b0cf0da4438507af91739beb7009211c1532a78f1e4da12beb6a615e64R1456) are final while these are not. can we make them consistent?